src/partykit/index.ts

```diff
@@ -49,3 +49,41 @@ export const connect: ConnectFunction = (
     return connection;
   });
 };
+
+async function getOrCreateRemoteName(dbName: string) {
+  const petnames = fireproof("petname.mappings");
+
+  try {
+    const doc = await petnames.get<{ remoteName: string; firstConnect: boolean }>(dbName);
+    return { remoteName: doc.remoteName, firstConnect: false };
+  } catch (_error) {
+    const remoteName = petnames.sthis.nextId().str;
+    await petnames.put({ _id: dbName, remoteName, firstConnect: true });
+    return { remoteName };
+  }
+}
+
+export function cloudConnect(
+  db: Database,
+  dashboardURI = BuildURI.from("https://dashboard.fireproof.storage/"),
+  partykitURL = "https://fireproof-cloud.jchris.partykit.dev/"
```

Comment:
For this, i would declare CoerceURI type which is able to receive 
String|Url|URI|BuildURI and with URI.from(....) is able to convert it to just one type.